tutorials/tree/dictionary/readTree.cpp

```diff
@@ -0,0 +1,51 @@
+// Author: Alvaro Tolosa-Delgado CERN 07/2023
+// Author: Jorge Agramunt Ros IFIC(Valencia,Spain) 07/2023
+
+/*************************************************************************
+ * Copyright (C) 1995-2023, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include <iostream>
+
+#include <TFile.h>
+#include <TTree.h>
+#include <TTreeReader.h>
+#include <TTreeReaderValue.h>
+
+#include "data2Tree.hpp"
+
+void readTree()
+{
+
+   std::unique_ptr<TFile> ifile = std::make_unique<TFile>("testFile.root", "read");
+   if ( nullptr == ifile ) {
+      std::cerr << " File not found " << std::endl;
+      return;
+   }
```

Comment:
Either:
```suggestion
   std::unique_ptr<TFile> ifile = std::make_unique<TFile>("testFile.root", "read");
   if ( !ifile || ifile->IsZombie() ) {
      std::cerr << " Problem opening the file testFile.root" << std::endl;
      return;
   }
```
Note that here the case (`!file, i.e ifile == nullptr`) will be extremely rare as a constructor can only fails to run if it can no allocate the memory.
```suggestion
   std::unique_ptr<TFile> ifile { TFile::Open("testFile.root", "read") };
   if ( !ifile || ifile->IsZombie() ) {
      std::cerr << " Problem opening the file testFile.root" << std::endl;
      return;
   }
```
which has the advantage of using the syntax that can open remote files.